root/edit/details/EditWork.js

```diff
@@ -0,0 +1,159 @@
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import DiffSide from '../../static/scripts/edit/components/edit/DiffSide';
+import FullChangeDiff from
+  '../../static/scripts/edit/components/edit/FullChangeDiff';
+import {commaOnlyListText} from
+  '../../static/scripts/common/i18n/commaOnlyList';
+
+type Props = {
+  ...EditT,
+  +data: {
+    new: {
+      attributes?: {
+        [attributeName: string]: $ReadOnlyArray<WorkAttributeT>,
+      },
+      languages: $ReadOnlyArray<string>,
+    },
+  },
+  +display_data: {
+    attributes?: {
+      [attributeName: string]: CompT<$ReadOnlyArray<string>>,
+    },
+    comment?: CompT<string | null>,
+    iswc?: CompT<string | null>,
+    language?: CompT<LanguageT | null>,
+    languages?: CompT<$ReadOnlyArray<string>>,
+    name?: CompT<string>,
+    type?: CompT<WorkTypeT | null>,
+    work: WorkT,
+  },
+};
+
+const EditWork = ({edit}: {edit: Props}) => {
+  const display = edit.display_data;
+  const comment = display.comment;
+  const iswc = display.iswc;
+  const name = display.name;
+  const language = display.language;
+  const languages = display.languages;
+  const attributes = display.attributes ?? {};
+  console.log(edit.data.new);
+  return (
+    <table className="details edit-work">
+      <tr>
+        <th>{addColonText(l('Work'))}</th>
+        <td colSpan="2">
+          <DescriptiveLink entity={display.work} />
+        </td>
+      </tr>
+      {name ? (
+        <WordDiff
+          label={addColonText(l('Name'))}
+          newText={name.new}
+          oldText={name.old}
+        />
+      ) : null}
+      {comment ? (
+        <WordDiff
+          label={addColonText(l('Disambiguation'))}
+          newText={comment?.new ?? ''}
+          oldText={comment?.old ?? ''}
+        />
+      ) : null}
+      {iswc ? (
+        <Diff
+          label={addColonText(l('ISWC'))}
+          newText={iswc?.new ?? ''}
+          oldText={iswc?.old ?? ''}
+        />
+      ) : null}
+      {display.type ? (
+        <FullChangeDiff
+          label={addColonText(l('Work type'))}
+          newContent={lp_attributes(display
+            .type?.new?.name ?? '', 'work_type')}
+          oldContent={lp_attributes(display
+            .type?.old?.name ?? '', 'work_type')}
+        />
+      ) : null}
+      {language ? (
+        <FullChangeDiff
+          label={addColonText(l('Language'))}
+          newContent={language.new?.name ?? ''}
+          oldContent={language.old?.name ?? ''}
+        />
+      ) : null}
+      {languages ? (
+        <Diff
+          label={addColonText(l('Lyrics Languages'))}
+          newText={commaOnlyListText(languages.new.map(
+            language => l_languages(language),
+          ))}
+          oldText={commaOnlyListText(languages.old.map(
+            language => l_languages(language),
+          ))}
+          split=", "
+        />
+      ) : null}
+      {edit.data.new.attributes
```

Comment:
I know it did check data in TT too, but i'm a bit confused why it would check that - we normally use only data_display. @mwiencek: do you see a clear reason we shouldn't use data_display for this?

Reply:
Nope, I agree we should just check `attributes`/`display.attributes` here. It should only exist if `data.new.attributes` exists already: https://github.com/metabrainz/musicbrainz-server/blob/2997b2a/lib/MusicBrainz/Server/Edit/Work/Edit.pm#L158-L159